include/zephyr/net/phy.h

```diff
@@ -232,6 +241,67 @@ static inline int phy_write(const struct device *dev, uint16_t reg_addr,
 	return api->write(dev, reg_addr, value);
 }
 
+/**
+ * @brief      Read PHY registers
+ *
+ * This routine provides a generic interface to read from a PHY register.
+ *
+ * @param[in]  dev       PHY device structure
+ * @param[in]  dev_addr  MMD device address
+ * @param[in]  reg_addr  Register address
+ * @param      value     Pointer to receive read value
+ *
+ * @retval 0 If successful.
+ * @retval -EIO If communication with PHY failed.
+ * @retval -ENOSYS if the interface is not implemented.
+ * @retval -EINVAL invalid device or register address.
+ */
+static inline int phy_read_device(const struct device *dev, uint8_t dev_addr, uint16_t reg_addr,
+				  uint16_t *value)
+{
+	const struct ethphy_driver_api *api = (const struct ethphy_driver_api *)dev->api;
+
+	if (api->read_device == NULL) {
+		return -ENOSYS;
+	}
+
+	if (dev_addr > 0x1F) {
```

Comment:
```suggestion
	if (dev_addr > MII_MMD_ACR_DEVAD_MASK) {
```
from `<zephyr/net/mii.h>`